### ISSUE TITLE
DeprecationWarning and Error on Ubuntu for iplot

### ISSUE THREAD

**Issue by paulamool** [CurtinIC/BioDendro#5]
DeprecationWarning and Error on Ubuntu for iplot
>>> iplot = tree.iplot(filename="simple_dendrogram.html", cutoff=0.4, width=900, height=500)
/home/ubuntu/.local/lib/python3.5/site-packages/plotly/graph_objs/_deprecations.py:426: DeprecationWarning:

plotly.graph_objs.Marker is deprecated.
Please replace it with one of the following more specific types
  - plotly.graph_objs.scatter.Marker
  - plotly.graph_objs.histogram.selected.Marker
  - etc.

/home/ubuntu/.local/lib/python3.5/site-packages/plotly/graph_objs/_deprecations.py:39: DeprecationWarning:

plotly.graph_objs.Data is deprecated.
Please replace it with a list or tuple of instances of the following types
  - plotly.graph_objs.Scatter
  - plotly.graph_objs.Bar
  - plotly.graph_objs.Area
  - plotly.graph_objs.Histogram
  - etc.

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ubuntu/BioDendro/BioDendro/cluster.py", line 389, in iplot
    dendro['data'].update({'hoverinfo': 'all'})
AttributeError: 'tuple' object has no attribute 'update'

**Comment by darcyabjones** [comment]
I'm not sure which error you were having trouble with.

I suspect that plotly has released a new version since we worked on it last.
The version that I was using was plotly==2.7.0 but the latest is now 3.1.1

It's possibly worth checking if their library includes those changes from your pull request.
(Am i misremembering that?).

The other option would be to specify the version of plotly to use in the setup.py file, or to update to work with their new datastructure.
It seems that a few of their changes have broken our plotting stuff.

Let me know if you want to go through it in person. 
I'm going to be super busy for the next few weeks though, so can't really afford to spend any time on it.

**Comment by paulamool** [comment]
Hi it's more the hoverinfo error.

Traceback (most recent call last):
File "", line 1, in 
File "/home/ubuntu/BioDendro/BioDendro/cluster.py", line 389, in iplot
dendro['data'].update({'hoverinfo': 'all'})
AttributeError: 'tuple' object has no attribute 'update'

**Comment by darcyabjones** [comment]
Ok. It seems that plotly has done a lot of work in that area in the last few months.
The data attribute of the graph_objs.Figure object used to be a custom type, but now it's a tuple

WIth plotly 2.7.0

>>> type(dendro["data"])
plotly.graph_objs.graph_objs.Data

With plotly 3.3.0

type(dendro["data"])

**Comment by darcyabjones** [comment]
Really the `create_dendro` function and `_Dendrogram` class need to be updated and the attributes that are getting edited in the iplot function need to be specified when you run `create_dendro`.

The other option could be to mimic the old update method.
E.g. in the `iplot` method of `Tree` the line:

```
dendro['data'].update({'hoverinfo': 'all'})
```

could become

```
d = []
for trace in dendro['data']:
    trace.update({'hoverinfo': 'all'})
    d.append(trace)

dendro['data'] = tuple(d)
```

It's a bit hacky, and likely t break next time plotly switches things around.
The simplest option is to just pin the plotly version so that it only install version 2.7.0.
Again, not ideal rewriting that tree plotting stuff would suck too.

**Comment by paulamool** [comment]
I think this might resolved for plotly release v3.4.0 
https://github.com/plotly/plotly.py/pull/1214
